src/internal_rep.rs

```diff
@@ -1545,6 +1565,370 @@ fn call_to_fc_rules<'a>(
+    let file_types = file_types_arg.get_list(context)?;
+
+    match fscontext_type {
+        FSContextType::XAttr | FSContextType::Task | FSContextType::Trans => {
+            // The 'regex_string_arg.get_range().is_none()' is a hacky way to
+            // to check if arg was actually provided or not.  Since we set the
+            // default for regex_string to "/" this is the only way I could find
+            // to test if the actual arg was passed or not
+            if regex_string_arg.get_range().is_none() && file_types.is_empty() {
+                ret.push(FileSystemContextRule {
+                    fscontext_type,
+                    fs_name,
+                    path: None,
+                    file_type: None,
+                    context: fs_context.clone(),
+                    file: file.clone(),
+                    // file_type_range shouldn't ever be used for xattr, task, or trans but I would rather not
+                    // have to deal with Option stuff later
+                    file_type_range: c.get_name_range().unwrap_or_default(),
+                    context_range: unwrap_or_return!(
+                        context_str_arg.get_range(),
+                        Err(CascadeErrors::from(InternalError::new()))
+                    ),
+                });
+            }
+            let mut errors = CascadeErrors::new();
+            if !file_types.is_empty() {
+                errors.append(CascadeErrors::from(
+                    ErrorItem::make_compile_or_internal_error(
+                        "File types can only be provided for 'genfscon'",
+                        Some(file),
+                        file_types_arg.get_range(),
+                        "",
+                    ),
+                ));
+            }
+            if regex_string_arg.get_range().is_some() {
+                errors.append(CascadeErrors::from(
+                    ErrorItem::make_compile_or_internal_error(
+                        "File path can only be provided for 'genfscon'",
+                        Some(file),
+                        regex_string_arg.get_range(),
+                        "",
+                    ),
+                ));
+            }
+            if !errors.is_empty() {
+                return Err(errors);
+            }
+        }
+        FSContextType::GenFSCon => {
+            if file_types.is_empty() {
+                ret.push(FileSystemContextRule {
+                    fscontext_type,
+                    fs_name,
+                    path: Some(regex_string),
+                    file_type: None,
+                    context: fs_context.clone(),
+                    file: file.clone(),
+                    // file_type_range shouldn't need to be used here since file_type is None, but I would rather not
+                    // have to deal with Option stuff later
+                    file_type_range: c.get_name_range().unwrap_or_default(),
+                    context_range: unwrap_or_return!(
```

Comment:
I think it would be more idiomatic to convert the Option into a Result using [ok_or_else](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.ok_or_else) and then use the ? operator.

`context_str_arg.get_range().ok_or_else(|| CascadeErrors::from(InternalError::new())?`